cli/azd/pkg/binding/linker_manager.go

```diff
@@ -0,0 +1,161 @@
+	return &linkerManager{
+		credentialProvider: credentialProvider,
+		httpClient:         httpClient,
+		armClientOptions:   armClientOptions,
+	}
+}
+
+type linkerManager struct {
+	credentialProvider account.SubscriptionCredentialProvider
+	httpClient         httputil.HttpClient
+	armClientOptions   *arm.ClientOptions
+}
+
+// Get a service linker resource
+func (sc *linkerManager) Get(
+	ctx context.Context,
+	subscriptionId string,
+	linkerConfig *LinkerConfig,
+) (*armservicelinker.LinkerResource, error) {
+	linkerClient, err := sc.createServiceLinkerClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := linkerClient.Get(ctx, linkerConfig.SourceResourceId, linkerConfig.Name, nil)
+	return &resp.LinkerResource, err
+}
+
+// Create a service linker resource
+func (sc *linkerManager) Create(
+	ctx context.Context,
+	subscriptionId string,
+	linkerConfig *LinkerConfig,
+) (*armservicelinker.LinkerResource, error) {
+	linkerClient, err := sc.createServiceLinkerClient(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	linkerResource := constructLinkerResource(linkerConfig)
+	poller, err := linkerClient.BeginCreateOrUpdate(
+		ctx, linkerConfig.SourceResourceId, linkerConfig.Name, linkerResource, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := poller.PollUntilDone(ctx, nil)
+	return &resp.LinkerResource, err
+}
+
+// Delete a service linker resource
+func (sc *linkerManager) Delete(
+	ctx context.Context,
+	subscriptionId string,
+	linkerConfig *LinkerConfig,
+) error {
+	linkerClient, err := sc.createServiceLinkerClient(ctx, subscriptionId)
+	if err != nil {
+		return err
+	}
+
+	poller, err := linkerClient.BeginDelete(ctx, linkerConfig.SourceResourceId, linkerConfig.Name, nil)
+	if err != nil {
+		return err
+	}
+
+	_, err = poller.PollUntilDone(ctx, nil)
+	return err
+}
+
+// Create a client to manage service linker resources
+func (sc *linkerManager) createServiceLinkerClient(
+	ctx context.Context,
+	subscriptionId string,
+) (*armservicelinker.LinkerClient, error) {
+	credential, err := sc.credentialProvider.CredentialForSubscription(ctx, subscriptionId)
+	if err != nil {
+		return nil, err
+	}
+
+	clientFactory, err := armservicelinker.NewClientFactory(credential, sc.armClientOptions)
+	if err != nil {
+		return nil, fmt.Errorf("creating service linker client: %w", err)
+	}
+
+	client := clientFactory.NewLinkerClient()
+	return client, nil
+}
+
+// Construct a resource payload used for linker resource creation
+func constructLinkerResource(
+	linkerConfig *LinkerConfig,
+) armservicelinker.LinkerResource {
+	// Fixed to use secret as auth type for azd
+	secretAuthType := armservicelinker.AuthTypeSecret
+	azureResourceType := armservicelinker.TargetServiceTypeAzureResource
+	secretTypeRawValue := armservicelinker.SecretTypeRawValue
+
+	return armservicelinker.LinkerResource{
+		Properties: &armservicelinker.LinkerProperties{
+			AuthInfo: &armservicelinker.SecretAuthInfo{
```

Comment:
opt-out auth and network?

Reply:
network should be opted-out, auth should be not. will update